script/tool/lib/src/publish_check_command.dart

```diff
@@ -63,6 +64,33 @@ class PublishCheckCommand extends PluginCommand {
     }
   }
 
+  Future<bool> hasValidPublishCheckRun(Directory package) async {
+    final io.Process process = await io.Process.start(
+      'flutter',
+      <String>['pub', 'publish', '--', '--dry-run'],
+      workingDirectory: package.path,
+    );
+
+    String output = '';
+    process.stdout.listen((List<int> event) {
+      io.stdout.add(event);
+      output += String.fromCharCodes(event);
+    });
+
+    process.stderr.listen((List<int> event) {
+      io.stderr.add(event);
+      output += String.fromCharCodes(event);
+    });
+
+    final int exitCode = await process.exitCode;
+
+    final bool hasPreReleaseSdk = output.contains('Package has 1 warning.') &&
```

Comment:
We need to wait on the output to ensure we're getting it all, to avoid potentially flaky results from the `contains` check. The `exitCode` docs say:

> There is no guarantee that stdout and stderr have finished reporting the buffered output of the process when the returned future completes. To be sure that all output is captured, wait for the done event on the streams.